controllers/cloud.redhat.com/providers/kafka/appinterface.go

```diff
@@ -69,9 +69,12 @@ func validateKafkaTopic(ctx context.Context, cl client.Client, nn types.Namespac
 	err := cl.Get(ctx, nn, &t)
 
 	if err != nil {
-		return &errors.MissingDependencies{
-			MissingDeps: map[string][]string{"topics": {nn.Name}},
-		}
+		missingDeps := errors.MakeMissingDependencies(errors.MissingDependency{
+			Source:  "kafka",
+			App:     nn.Name,
+			Details: fmt.Sprintf("No topics found for app '%s' found in namespace '%s'", nn.Name, nn.Namespace),
```

Comment:
It looks like we tried to `Get` a specific KafkaTopic resource up above -- is this description better?

```suggestion
			Details: fmt.Sprintf("No KafkaTopic named '%s' found in namespace '%s'", nn.Name, nn.Namespace),
```